akka-cluster-tools/src/main/resources/reference.conf

```diff
@@ -30,8 +30,8 @@ akka.cluster.pub-sub {
   max-delta-elements = 3000
 
   # When a message is published to a topic with no subscribers send it to the dead letters.
-  send-to-dead-letters-when-no-subscribers = on
-  
+  send-to-dead-letters-when-no-subscribers = false
```

Comment:
When it was introduced in #23462 the default `on` was probably chosen to keep the same behavior as before, i.e. send to deadletters. We could debate which is the best default but it's probably good for many to be able to see the dead letters rather than complete silence, and then it's possible to opt out of it.

Also, we should probably not change such behavior in a patch release.

Reply:
I should have missed something during my tests. I actually didn't see any dead letters while my subscribers list was empty. It made me think that `on` was an invalid value for a config boolean. I didn't investigate enough and didn't read the `DefaultTransformer` implementation. Sorry for that. I will try to reproduce what I noticed.

Reply:
@patriknw thanks for your review. I can't understand why I didn't see the dead letters messages during my former tests, but I can't reproduce now... I will close this PR.